### ISSUE TITLE
Can't register new users, even with `registration_shared_secret` configured

### ISSUE THREAD

**Issue by bbabel** [matrix-org/synapse#9894]
Hello,

I just installed my synapse server and everything seems to work fine. I even tested federation on https://federationtester.matrix.org/ and it works, but when I try to register a new user with `register_new_matrix_user -c homeserver.yaml http://localhost:8008` I got:

```
Sending registration request...
ERROR! Received 400 Bad Request
Shared secret registration is not enabled
```
I checked my homeserver.yaml and I can see this line: `registration_shared_secret: "[my secret]"`. I've tried with and without quotes but got the same response.

I also changed `enable_registration: true` without success.

Am I missing something? Thank you.

**Comment by dklimpel** [comment]
Are you sure that your prompt is in your config path? In other words: Is the path to your homeserver.yaml right?

**Comment by bbabel** [comment]
> Are you sure that your prompt is in your config path? In other words: Is the path to your homeserver.yaml right?

I think it is, because I do get this kind of response:

```
New user localpart: erikj
Password:
Confirm password:
Make admin [no]:
```
Just that in the last step instead of `Success!` I get `Shared secret registration is not enabled`.

**Comment by anoadragon453** [comment]
"Shared secret registration is not enabled" will appear if `registration_shared_secret` is falsey or absent from the config file.

Are you sure you don't have another instance of `registration_shared_secret: false` in your config file somewhere?

It looks like you may be using the Debian package, which tends to use multiple files in `/etc/matrix-synapse` to store homeserver configuration that eventually get merged together into one config.

**Comment by bbabel** [comment]
I typed `systemctl status matrix-synapse` and confirmed Synapse was running, but I noticed a warning about `macaroon_secret_key`, a string in the config file that was commented out.

After activate that string synapse refused to run, so I commented it out again and it was then when `systemctl status matrix-synapse` (which didn't activate) showed me an error in the config that I haven't seen before.

After comment out that error Synapse was able to run and register the user.

I'm not really sure what happened, but that solved my problem. Thank you for your help.
